src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AzureManagedIdentityAuthenticationProvider.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClient
+{
+    internal sealed class AzureManagedIdentityAuthenticationProvider : SqlAuthenticationProvider
+    {
+        // HttpClient is intended to be instantiated once and re-used throughout the life of an application.
+#if NETFRAMEWORK
+        private static readonly HttpClient s_httpClient = new HttpClient();
+#else
+        private static readonly HttpClient s_httpClient = new HttpClient(new HttpClientHandler() { CheckCertificateRevocationList = true });
+#endif
+
+        private const string AzureSystemApiVersion = "&api-version=2019-08-01";
+        private const string AzureVmImdsApiVersion = "&api-version=2018-02-01";
+        private const string AccessToken = "access_token";
+        private const string Expiry = "expires_on";
+        private const int FileTimeLength = 10;
+
+        private const int DefaultRetryTimeout = 0;
+        private const int DefaultMaxRetryCount = 5;
```

Comment:
Any reason we pick this max retry count?

Reply:
Based on Retry guidance:
https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance